src/v/kafka/protocol/kafka_batch_adapter.cc

```diff
@@ -62,7 +62,8 @@ model::record_batch_header kafka_batch_adapter::read_header(iobuf_parser& in) {
     auto header = model::record_batch_header{
       .size_bytes = size_bytes,
       .base_offset = base_offset,
-      .type = raft::data_batch_type,
+      .type = attrs.is_control() ? raft::configuration_batch_type
```

Comment:
great find. 
cc: @dotnwat

Reply:
I'm not immediately understanding what this patch is doing. IIRC `kafka_batch_adapter::read_header` is parsing kafka batches, so I don't think it would ever be the case that a client sent us a batch with type `raft::configuration_batch_type`, and there are other types of control batches (e.g. transactions).

can you explain a bit more the motivation for the patch?

Reply:
The `partition_response::record_set::consume_batch` returns a `kafka_batch_adapter`; prior to this change it's non-obvious how to filter config batches from the response, since the `record_batch_header::type` was incorrectly set.

Reply:
The control flag is set on the batches in a fetch response (see `fetch.cc:adapt_fetch_batch`) so I think you can check that. ultimately the proxy should act as a kafka client, and kafka batches don't have a `type` property like we do internally. but.. am I missing something?

Reply:
I'm not convinced that a client of the proxy is concerned with a config batch, nor am I sure how to represent it to the client.

There is no `type` property, but `attr[5]` is specified here: https://kafka.apache.org/documentation/#recordbatch.

I thought it made sense to filter it, but I specifically left it off of #1154 because I thought it might be contentious.

Reply:
It does make sense that the proxy should filter out non-data batches. But I'm still trying to figure out what this change accomplishes and why, and if this is the right approach. There isn't any information in the commit message, so I'm largely guessing and trying to piece together the answer from other parts of the PR.

Reply:
Rebased with comments

Reply:
It's entirely possible that `kafka_batch_adapter` was always the wrong tool for the job, and that `adapt_fetch_batch` does the right thing.

Reply:
actually noah has a good point, shouldn't we do the same as https://github.com/Shopify/sarama/pull/1898/files on the kafka::client so that the http handler wouldn't have to deal w/ it. i mean, it's filtered in the lower levels

Reply:
I'm not sure what's being suggested, so I've rebased this PR to just filter based on `is_control` in the proxy.

An alternative is to filter them in the client: https://github.com/vectorizedio/redpanda/compare/dev...BenPope:pandaproxy-drop-config-batches-2

Reply:
the suggestion was to push this to the `kafka::client` level. i thought pandaproxy just uses the `kafka::client`. 

What i'm saying is that most upstream clients do the filtering of these control batches at the *client* level, not at the application level.

Reply:
@BenPope as long as it isn't a big change, I think filtering in the client itself is a better solution since no application using the client would ever want these control batches. plus, we'll have to deal with transaction control batches later in the client in any case. ultimately its mostly a non-issue once Michal's changes land that create 1-1 offset compat layer with kafka.

Reply:
Yes, but looking at it I like the version in this PR where you examine the control batch above the client level. Making the change in the client is ideal, but I didn't realize how low-level our client is: it returns raw fetch responses. So in order to do it properly in the client we'd need to either re-write the responses or add an interface above the low level fetch. I think the later will have to be done at some point, but by that time this control batch filtering will be irrelevant anyway.